test/unit/getTickFormat.spec.js

```diff
@@ -0,0 +1,667 @@
+      'xFakeDomain': [0, 100, 200]
+    };
+
+    data.min = _.min(opts.xDomain);
+    data.max = _.max(opts.xDomain);
+
+    _.each(opts.xFakeDomain, function(v) {
+      ticks.push(util.getTickFormat(v, logScale, data, opts));
+    });
+
+    // console.log(opts.xDomain, opts.xFakeDomain, ticks);
+
+    expect(ticks).to.have.lengthOf(opts.xDomain.length); // same length
+    for (var i = 0; i < opts.xDomain.length - 1; i++) {
+      expect(ticks[i]).to.be.equal(opts.xDomain[i]);
+    }
+  });
+
+  it('8.1.1 Small Data: initial of ticks is "1", no NA and noGrouping = false', function() {
+    ticks = [];
+    logScale = false;
+    data = {
+      'noGrouping': false,
+      'hasNA': false,
+      'smallDataFlag': true,
+      'min': '',
+      'max': ''
+    };
+
+    opts = {
+      'xDomain' : [0.00000000001, 0.0000000001, 0.000000001, 0.00000001, 0.0000001, 0.000001, 0.00001, 0.0001, 0.001],
+      'emptyMappingVal' : ''
+    };
+
+    data.min = _.min(opts.xDomain);
+    data.max = _.max(opts.xDomain);
+
+    _.each(opts.xDomain, function(v) {
+      ticks.push(util.getTickFormat(v, logScale, data, opts));
+    });
+
+    // console.log(opts.xDomain, ticks);
+
+    expect(ticks).to.have.lengthOf(opts.xDomain.length); // same length
+  });
+
+  it('8.1.2 Small Data: initial of ticks is not "1", no NA and noGrouping = false\'', function() {
+    ticks = [];
+    logScale = false;
+    data = {
+      'noGrouping': false,
+      'hasNA': false,
+      'smallDataFlag': true,
+      'min': '',
+      'max': ''
+    };
+
+    opts = {
+      'xDomain' : [0.00000000006, 0.0000000001, 0.0000000006, 0.000000001, 0.000000006, 0.00000001, 0.00000006],
+      'emptyMappingVal' : ''
+    };
+
+    data.min = _.min(opts.xDomain);
+    data.max = _.max(opts.xDomain);
+
+    _.each(opts.xDomain, function(v) {
+      ticks.push(util.getTickFormat(v, logScale, data, opts));
+    });
+
+    // console.log(opts.xDomain, ticks);
+
+    expect(ticks).to.have.lengthOf(opts.xDomain.length); // same length
+  });
+
+  it('8.2.1 Small Data: initial of ticks is "1", has NA and noGrouping = false', function() {
+    ticks = [];
+    logScale = false;
+    data = {
+      'noGrouping': false,
+      'hasNA': true,
+      'smallDataFlag': true,
+      'min': '',
+      'max': ''
+    };
+
+    opts = {
+      'xDomain' : [0.00000000001, 0.0000000001, 0.000000001, 0.00000001, 0.0000001, 0.000001, 0.00001, 0.0001, 0.001],
+      'emptyMappingVal' : 0.001
+    };
+
+    data.min = _.min(opts.xDomain);
+    data.max = _.max(opts.xDomain);
+
+    _.each(opts.xDomain, function(v) {
+      ticks.push(util.getTickFormat(v, logScale, data, opts));
+    });
+
+    // console.log(opts.xDomain, ticks);
+
+    expect(ticks).to.have.lengthOf(opts.xDomain.length); // same length
+    expect(ticks[ticks.length - 1]).to.be.equal('NA'); // last tick = 'NA' 
+  });
+
+  it('8.2.2 Small Data: initial of ticks is not "1", has NA and noGrouping = false\'', function() {
```

Comment:
What's the difference between 8.2.1 and 8.2.2 except the xDomains are different? If you are trying to test the algorithm can accept the different input, could you give some more information how the input is designed? Why the test is needed?